official/recommendation/ncf_estimator_benchmark.py

```diff
@@ -0,0 +1,161 @@
+    default_flags['epsilon'] = 1.45439e-07
+    default_flags['layers'] = [256, 256, 128, 64]
+    default_flags['num_factors'] = 64
+    default_flags['hr_threshold'] = 0.635
+    default_flags['ml_perf'] = True
+    default_flags['use_synthetic_data'] = False
+    default_flags['data_dir'] = os.path.join(
+        root_data_dir,
+        benchmark_base.NCF_DATA_DIR_NAME)
+
+    super(KerasNCFRealData, self).__init__(
+        ncf_estimator_main.run_ncf,
+        output_dir=output_dir,
+        default_flags=default_flags,
+        **kwargs)
+
+  def _extract_benchmark_report_extras(self, stats):
+    metrics = []
+    metrics.append({'name': 'exp_per_second',
+                    'value': stats['avg_exp_per_second']})
+
+    # Target is 0.625, but some runs are below that level. Until we have
+    # multi-run tests, we have to accept a lower target.
+    metrics.append({'name': 'hr_at_10',
+                    'value': stats['eval_hit_rate'],
+                    'min_value': 0.618,
+                    'max_value': 0.635})
+
+    metrics.append({'name': 'train_loss',
+                    'value': stats['loss']})
+
+    return metrics
+
+  def benchmark_1_gpu(self):
+    self._setup()
+    self._run_and_report_benchmark()
+
+  def benchmark_1_gpu_no_cloning(self):
+    self._setup()
+    FLAGS.clone_model_in_keras_dist_strat = False
+    self._run_and_report_benchmark()
+
+  def benchmark_2_gpus(self):
+    self._setup()
+    FLAGS.num_gpus = 2
+    self._run_and_report_benchmark()
+
+  def benchmark_2_gpus_no_cloning(self):
+    self._setup()
+    FLAGS.num_gpus = 2
+    FLAGS.clone_model_in_keras_dist_strat = False
+    self._run_and_report_benchmark()
+
+
+class KerasNCFSyntheticData(benchmark_base.KerasNCFBenchmarkBase):
+  """Benchmark NCF model using synthetic data."""
+
+  def __init__(self,
+               output_dir=None,
+               default_flags=None,
+               **kwargs):
+
+    default_flags = {}
+    default_flags['dataset'] = 'ml-20m'
+    default_flags['num_gpus'] = 1
+    default_flags['train_epochs'] = 14
+    default_flags['batch_size'] = 160000
+    default_flags['learning_rate'] = 0.00382059
+    default_flags['beta1'] = 0.783529
+    default_flags['beta2'] = 0.909003
+    default_flags['epsilon'] = 1.45439e-07
+    default_flags['layers'] = [256, 256, 128, 64]
+    default_flags['num_factors'] = 64
+    default_flags['hr_threshold'] = 0.635
+    default_flags['use_synthetic_data'] = True
+
+    super(KerasNCFSyntheticData, self).__init__(
+        ncf_estimator_main.run_ncf,
+        output_dir=output_dir,
+        default_flags=default_flags,
+        **kwargs)
+
+  def _extract_benchmark_report_extras(self, stats):
+    metrics = []
+    metrics.append({'name': 'exp_per_second',
+                    'value': stats['avg_exp_per_second']})
+    return metrics
+
+  def benchmark_1_gpu(self):
+    self._setup()
+    self._run_and_report_benchmark()
+
+  def benchmark_1_gpu_no_cloning(self):
```

Comment:
Does this test make sense for estimator? I did not look but does the estimator version accept the no_cloning option.